File: mg01/game/tech/Assets/Script/Org/Action/Instant/ActionInstant.cs
/* ==============================================================================
 * 即时动作
 * @author jr.zeng
 * 2016/10/28 11:25:00
 * ==============================================================================*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using UnityEngine;

namespace mg.org.Actions
{

    public class ActionInstant : ActionBase
    {
        public ActionInstant()
        {

        }

        public void Init()
        {
            m_inited = true;
        }

        public override void Step(float dt_)
        {
            Progress(1);
        }


        public override void Progress(float progress_)
        {
            if (progress_ >= 1)
            {
                Done();
            }
        }
        

    }


}

File: mg01/game/tech/Assets/Resources/LuaScript/src/org/action/instant/ActionInstant.lua
--ActionInstant
--@author jr.zeng
--2017年7月17日 下午3:54:19
local modname = "ActionInstant"
--==================global reference=======================

--===================namespace========================
local ns = "org.action"
local using = {--[[namespace1,...]]}
local _ENV = namespace(ns, using)
--===================module property========================
local super = ActionBase	--父类
_ENV[modname] = class(modname, super)
local ActionInstant = _ENV[modname]

--===================module content========================

function ActionInstant:__ctor()
    
end

function ActionInstant:init()
    
    self.m_inited = true
end


--override
function ActionInstant:update(dt)

    self:progress(1)
end


--override
function ActionInstant:progress(value)
    
    if value >= 1 then
        self:done()
    end
end

return ActionInstant